src/python/RestClient/AuthHandling/X509Auth.py

```diff
@@ -82,16 +86,30 @@ def __search_cert_key(self):
         else:
             raise ClientAuthException("No valid X509 cert-key-pair found.")
 
+    def __create_ca_bundle(self):
+        trust_ca_files = glob.glob(f"{self._ca_path}/*.pem")
+        trust_ca_files.append(self._ssl_cert)
+        ca_bundle = f"/tmp/trust_ca_{os.getuid()}.crt"
+
+        with open(ca_bundle,"wb") as bundles:
+            for trust_ca in trust_ca_files:
+                with open(trust_ca,'rb') as f:
+                    bundles.write(f.read())
+
+        return ca_bundle
+
     def configure_auth(self, curl_object):
         curl_object.setopt(curl_object.SSL_VERIFYPEER, self._ssl_verifypeer)
         #not sure if CAPATH shoud be set. YG 2021-Oct-11
         curl_object.setopt(curl_object.CAPATH, self._ca_path)
         curl_object.setopt(curl_object.SSLCERT, self._ssl_cert)
         curl_object.setopt(curl_object.SSLKEY, self._ssl_key)
+
+        if self._ssl_proxy_ca:
+            self._ca_info = self.__create_ca_bundle()

```

Comment:
If I understand the logic right then `self._ca_info` will be created over and over. What is missing that the code should check first if `self._ca_info` exists and only perform this operation once. This will initialize `self._ca_info` once and allow code to re-use it on future requests.

Reply:
I updated a commit to resolve this request. The commit will check whether ```self._ca_info``` is set or not. Then, it will create a temporary ca info cert file.